lib/rack/lint.rb

```diff
@@ -134,10 +134,15 @@ def response
       ##
       ## Incoming HTTP requests are represented using an environment. \
       def check_environment(env)
-        ## The environment must be an unfrozen instance of +Hash+. The Rack application is free to modify the environment, but the modified environment should also comply with this specification.
+        ## The environment must be an unfrozen instance of +Hash+. The Rack application is free to modify the environment, but the modified environment should also comply with this specification. \
         raise LintError, "env #{env.inspect} is not a Hash, but #{env.class}" unless env.kind_of? Hash
         raise LintError, "env should not be frozen, but is" if env.frozen?
 
+        ## All environment keys must be strings.
+        unless env.keys.all?(String)
+          raise LintError, "env contains non-string keys"
```

Comment:
This error message does not report which keys are problematic. As a user, I could imagine it would be hard to find which key is not a String. Can we log that?

Reply:
What about something like:

```ruby
keys = env.keys.reject{|key| String === key}
if keys.any?
  raise LintError, "env contains non-string keys: #{keys.inspect}"
end
```

Reply:
Pushed another commit for that.